### DISCUSSION TITLE
Make collections and documentation consistent

### DISCUSSION THREAD

**Discussion by dkkb** [outline/outline#2904] [Ideas]
Hi, @tommoor. Happy New Year.

Here's the situation: right now, collection and document are two separate concepts, although it's actually better and more consistent to combine them.

We truly want the same function on them in a lot of scenarios.
e.g. 

- We have the option of selecting a collection icon, but what about a document icon?
- We can use the sidebar button to create a new nested document, but we can't create a new document in collection just one click. Related to https://github.com/outline/outline/pull/2864
- We can now manage this collection's permissions, but not the permissions in documents. It can be beneficial to a large corporation at times.
- We can drag and drop the document into another document and make it nested, but not the collection.

Actually, I used Notion and other products, and they just use documents to replace collections. That's really really simple for the users and developers.

<img src="https://user-images.githubusercontent.com/82504881/148377103-fe5f8a59-d6e1-4051-8f16-1d8712b9ccad.png" width=300>

<img src="https://user-images.githubusercontent.com/82504881/148379208-7c41b45b-4f07-42e3-a4a9-3105c7ae11f4.png" width=300>

**Comment by tommoor** [comment]
It sounds like you want Outline to be a copy of Notion – this is not our intent. While there are certainly places where we can bring collections and document functionality more inline it's unlikely these concepts will ever be combined in Outline, it would mean essentially rewriting the entire project at this point.

> we can't create a new document in collection just one click

This is fair – we should find a way to fit this in.

> We can drag and drop the document into another document and make it nested, but not the collection.

This should work!

> permissions in documents

Some permissions are on the roadmap, but they are likely to be simpler than collections.

**Reply by dkkb** [reply]
In fact, the document management system I use at work combines these concepts,too. What I want isn't just a Notion copy, because I don't like many of the designs in Notion, but I think combining them is incredibly helpful for users and developers. If you want to add new features to `document`, you may not need to do it again on the `collection`. In the long term, I believe this will greatly aid the system's design simplicity. Just some suggestion.

**Reply by dkkb** [reply]
There're many fantastic features in sidear that I'd like to bring over to `collection`.

For example, we can expand nested documents so that we can quickly discover the next document to edit, especially if we have a lot of them. (Sometimes we categorize documents, although there are not many in each category.)
These behaviors are fantastic, and I'd want to see them applied on `collection`.

<img src="https://user-images.githubusercontent.com/82504881/148668226-d035f249-99bb-42c0-b18e-a6ab590b8648.png" width=200>
